lib/rhea-promise/index.ts

```diff
@@ -0,0 +1,662 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import * as rhea from "rhea";
+import * as debugModule from "debug";
+
+const debug = debugModule("rhea-promise");
+
+/**
+ * Eastablishes an amqp connection.
+ * @param {ConnectionOptions} [options] Options to be provided for establishing an amqp connection.
+ * @return {Promise<Connection>} Promise<Connection>
+ * - **Resolves** the promise with the Connection object when rhea emits the "connection_open" event.
+ * - **Rejects** the promise with an AmqpError when rhea emits the "connection_close" event while trying
+ * to establish an amqp connection.
+ */
+export async function connect(options?: ConnectionOptions): Promise<any> {
+  return new Promise((resolve, reject) => {
+    const connection = rhea.connect(options);
+
+    function onOpen(context: any): void {
+      connection.removeListener("connection_open", onOpen);
```

Comment:
nit: may be worth moving the `removeListener` calls into a separate function so that they don't get duplicated in the success/failure cases in each one of these